database/certificate.go

```diff
@@ -665,3 +665,56 @@ func (db *DB) IsTrustValid(id int64) (bool, error) {
 	err := row.Scan(&isValid)
 	return isValid, err
 }
+
+// GetAllCertsInChain returns the certificate chain of which tipCertID is the tip of,
+// which is to say it returns the certificate that trusts tipCertID, and the
+// certificate that trusts that other certificate, etc.
+func (db *DB) GetAllCertsInChain(tipCertID int64) ([]certificate.Certificate, error) {
+	rows, err := db.Query(`
+WITH RECURSIVE parents AS (
+	SELECT * FROM trust WHERE cert_id=$1
```

Comment:
I'm guessing it will, because I'm not sure what happens when you call `Scan` and get more rows than expected. We should probably also store all those columns in a variable somewhere instead of repeating them in every query, because we use the exact same fields in several functions. In fact, the scanCert function expects those fields.